### PR TITLE
Prepare for FxA payments server moving to our GCPv2 setup (OPST-296)

### PR DESCRIPTION
**Pull request by sean-rose**
After which its production logs should end up in `moz-fx-fxa-prod.gke_fxa_prod_log.stderr` or `moz-fx-fxa-prod.gke_fxa_prod_log.stdout`, and its stage logs should end up in `moz-fx-fxa-nonprod.gke_fxa_stage_log.stderr` or `moz-fx-fxa-nonprod.gke_fxa_stage_log.stdout`.

See https://mozilla-hub.atlassian.net/browse/OPST-296?focusedCommentId=821241.

---
Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.
- [ ] When adding a new derived dataset, ensure that data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data can be available in the [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](https://github.com/mozilla/bigquery-etl/blob/main/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/DENG-2512)

### CODE REVIEW

File: sql/moz-fx-data-shared-prod/firefox_accounts_derived/fxa_gcp_stdout_events_v1/query.sql [hunk 1]
```diff
@@ -25,11 +25,11 @@ WHERE
   )
   AND DATE(`timestamp`) = @submission_date
   AND jsonPayload.type = 'amplitudeEvent'
-  -- We expect to only see events from fxa-auth-server and fxa-content-server here
+  -- We expect to only see events from fxa-auth-server, fxa-content-server, and fxa-payments-server here.
```
Comment (akkomar) [inline sql/moz-fx-data-shared-prod/firefox_accounts_derived/fxa_gcp_stdout_events_v1/query.sql:28]: This line is confusing. According to https://mozilla-hub.atlassian.net/browse/FXA-8315
`fxa-auth-server` goes to `stderr`.
(same in stderr query)

**Review by akkomar — APPROVED** [review]

File: sql/moz-fx-data-shared-prod/firefox_accounts_derived/fxa_gcp_stdout_events_v1/query.sql (hunk 1 above)
Comment (sean-rose) [reply sql/moz-fx-data-shared-prod/firefox_accounts_derived/fxa_gcp_stdout_events_v1/query.sql:28]: These comments were originally written by you and @kik-kik in #4288.  I think the following three comment lines try to clarify that this is to future-proof these ETLs for when FxA standardizes their logging to either all go to `stdout` or all go to `stderr`.

Maybe this wording change would help reduce confusion?
```suggestion
  -- We might see events from fxa-auth-server, fxa-content-server, or fxa-payments-server here.
```

Comment (akkomar) [reply sql/moz-fx-data-shared-prod/firefox_accounts_derived/fxa_gcp_stdout_events_v1/query.sql:28]: Yes, that sounds better.

File: sql/moz-fx-data-shared-prod/firefox_accounts_derived/fxa_gcp_stderr_events_v1/query.sql [hunk 2]
```diff
@@ -25,11 +25,11 @@ WHERE
   )
   AND DATE(`timestamp`) = @submission_date
   AND jsonPayload.type = 'amplitudeEvent'
-  -- We expect to only see events from fxa-auth-server and fxa-content-server here
+  -- We expect to only see events from fxa-auth-server, fxa-content-server, and fxa-payments-server here.
```
Comment (akkomar) [inline sql/moz-fx-data-shared-prod/firefox_accounts_derived/fxa_gcp_stderr_events_v1/query.sql:28]:
```suggestion
  -- We might see events from fxa-auth-server, fxa-content-server, or fxa-payments-server here.
```

File: sql/moz-fx-data-shared-prod/firefox_accounts_derived/nonprod_fxa_gcp_stdout_events_v1/query.sql
Comment (kik-kik) [inline sql/moz-fx-data-shared-prod/firefox_accounts_derived/nonprod_fxa_gcp_stdout_events_v1/query.sql:1]: Should we include a note in the metadata.yaml description to indicate that the payments events only started arriving via this query after xxxx.xx.xx?

**Review by kik-kik — APPROVED** [review]
Looks like this should work without any further changes. The only comment I have is that maybe we should include a note about the payments data only being available in the "new" GCP fxa log tables after a specific date and prior to it the events were arriving in the stdout table.

Also, I wonder if we should include a `<` date filter in the fxa_all_events view to further indicate this:
https://github.com/mozilla/bigquery-etl/blob/6c185d0a88b430d795387a3b9837a73899b46064/sql/moz-fx-data-shared-prod/firefox_accounts/fxa_all_events/view.sql#L123

File: sql/moz-fx-data-shared-prod/firefox_accounts_derived/nonprod_fxa_gcp_stdout_events_v1/query.sql
Comment (sean-rose) [reply sql/moz-fx-data-shared-prod/firefox_accounts_derived/nonprod_fxa_gcp_stdout_events_v1/query.sql:1]: While FxA stage payments server logs started appearing in this source on Jan 4, it's unclear to me what's actually going on with that (whether that's the existing payments server and/or [vNext](https://mozilla-hub.atlassian.net/browse/OPST-1078), and there didn't appear to be any corresponding drop in the FxA stage payments server logs from AWS), so I'd prefer not to add incomplete or potentially misleading comments at this point.

Once the migration is complete and we know what actually happened I'd be in favor of adding clarifying comments.

I've also [asked the FxA team for an update in Slack](https://mozilla.slack.com/archives/C4D36CAJW/p1709162582331549).

**Comment by sean-rose** [comment]
> Looks like this should work without any further changes. The only comment I have is that maybe we should include a note about the payments data only being available in the "new" GCP fxa log tables after a specific date and prior to it the events were arriving in the stdout table.
> 
> Also, I wonder if we should include a `<` date filter in the fxa_all_events view to further indicate this:
> 
> https://github.com/mozilla/bigquery-etl/blob/6c185d0a88b430d795387a3b9837a73899b46064/sql/moz-fx-data-shared-prod/firefox_accounts/fxa_all_events/view.sql#L123

The migration hasn't happened in production yet, but once that's been done I'd be in favor of both adding comments in the table descriptions and including date filters where that's feasible.

**Pull request merged**